Src/FluentAssertions/Equivalency/EnumEqualityStep.cs

```diff
@@ -56,5 +52,59 @@ public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAsserti
 
             return true;
         }
+
+        private static void HandleByValue(IEquivalencyValidationContext context)
+        {
+            decimal? subjectsUnderlyingValue = ExtractDecimal(context.Subject);
+            decimal? expectationsUnderlyingValue = ExtractDecimal(context.Expectation);
+
+            string subjectsName = EnumDescription(context.Subject, subjectsUnderlyingValue);
+            string expectationName = EnumDescription(context.Expectation, expectationsUnderlyingValue);
+
+            Execute.Assertion
+                .ForCondition(subjectsUnderlyingValue == expectationsUnderlyingValue)
+                .FailWith($"Expected enum to be {expectationName}{{reason}}, but found {subjectsName}.");
```

Comment:
❓ How about something like `Expected enum to equal {expectationName} by name{{reason}}, but found {subjectsName}.`